### PR TITLE
restructure TPCFastTransform

### PR DESCRIPTION
**Pull request by sgorbuno**
restructure TPCFastTransform;
consistency tests;
inverse transformation;
extend the area of SP distortion approximation to the sector edges
get rid of unique_ptr in the GPU code

### CODE REVIEW

**Comment by davidrohr** [comment]
@sgorbuno : Is this ready from your side? Then I'll double check that it doesn't break AliRoot and whether OpenCL works fine now, and then merge it.

File: GPU/TPCFastTransformation/TPCFastTransformGeo.h [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+/// \file  TPCFastTransformGeo.h
+/// \brief Definition of TPCFastTransformGeo class
+///
+/// \author  Sergey Gorbunov <sergey.gorbunov@cern.ch>
+
+#ifndef ALICEO2_GPUCOMMON_TPCFASTTRANSFORMATION_TPCFASTTRANSFORMGEO_H
+#define ALICEO2_GPUCOMMON_TPCFASTTRANSFORMATION_TPCFASTTRANSFORMGEO_H
+
+#include "GPUCommonDef.h"
+#include <memory>
```
Comment (davidrohr) [inline GPU/TPCFastTransformation/TPCFastTransformGeo.h]: Please protect `#include <memory>` with `#ifndef __OPENCL__`

**Comment by sgorbuno** [comment]
@davidrohr, 
How do you compile the O2/GPU/* code in AliRoot? 
I tried to copy the code to AliTPCCommon, but AliRoot does not compile with AliTPCComon/master  
is there a trick?

**Comment by davidrohr** [comment]
@sgorbuno : Please copy the full directory O2/GPU to AliRoot/GPU.
It might be that it is already broken in the o2 dev branch. Please use https://github.com/davidrohr/AliceO2/commits/dev as basis, which contains the fix (will open a PR soon).

**Comment by davidrohr** [comment]
@sgorbuno : Attached a patch with few more fixes
[sergey.patch.gz](https://github.com/AliceO2Group/AliceO2/files/3326450/sergey.patch.gz)

**Comment by davidrohr** [comment]
@sgorbuno : btw AliRoot compilation should be fixed in O2/dev. If you rebase this PR to dev branch, and then copy over the GPU folder to AliRoot, you should be able to reproduce the errors I pasted above.

**Pull request merged**